// regexp-worker/lib/Procedures/procExecRegExp.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createResponseExecRegExp = exports.createRequestExecRegExp = exports.procExecRegExp = exports.isExecRegExpResponse = exports.isExecRegExpRequest = exports.requestTypeExecRegExp = void 0;
const procedure_1 = require("./procedure");
const evaluateRegExp_1 = require("../helpers/evaluateRegExp");
exports.requestTypeExecRegExp = 'ExecRegExp';
exports.isExecRegExpRequest = procedure_1.genIsRequest(exports.requestTypeExecRegExp);
exports.isExecRegExpResponse = procedure_1.genIsResponse(exports.requestTypeExecRegExp);
function procExecRegExp(r) {
    if (!exports.isExecRegExpRequest(r))
        return undefined;
    try {
        const regex = evaluateRegExp_1.toRegExp(r.data.regexp);
        return createResponseExecRegExp(r, evaluateRegExp_1.execRegExp(regex, r.data.text));
    }
    catch (e) {
        return procedure_1.createErrorResponse(r, e.toString());
    }
}
exports.procExecRegExp = procExecRegExp;
function createRequestExecRegExp(data) {
    return procedure_1.createRequest(exports.requestTypeExecRegExp, data);
}
exports.createRequestExecRegExp = createRequestExecRegExp;
function createResponseExecRegExp(request, data) {
    return procedure_1.createResponse(request.id, request.requestType, data);
}
exports.createResponseExecRegExp = createResponseExecRegExp;
