// regexp-worker/lib/Procedures/procMatchRegExp.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createResponseMatchRegExp = exports.createRequestMatchRegExp = exports.procMatchRegExp = exports.isMatchRegExpResponse = exports.isMatchRegExpRequest = exports.requestTypeMatchRegExp = void 0;
const procedure_1 = require("./procedure");
const evaluateRegExp_1 = require("../helpers/evaluateRegExp");
exports.requestTypeMatchRegExp = 'MatchRegExp';
exports.isMatchRegExpRequest = procedure_1.genIsRequest(exports.requestTypeMatchRegExp);
exports.isMatchRegExpResponse = procedure_1.genIsResponse(exports.requestTypeMatchRegExp);
function procMatchRegExp(r) {
    if (!exports.isMatchRegExpRequest(r))
        return undefined;
    try {
        const regex = evaluateRegExp_1.toRegExp(r.data.regexp);
        const regexResult = evaluateRegExp_1.matchRegExp(r.data.text, regex);
        return createResponseMatchRegExp(r, regexResult);
    }
    catch (e) {
        return procedure_1.createErrorResponse(r, e.toString());
    }
}
exports.procMatchRegExp = procMatchRegExp;
function createRequestMatchRegExp(data) {
    return procedure_1.createRequest(exports.requestTypeMatchRegExp, data);
}
exports.createRequestMatchRegExp = createRequestMatchRegExp;
function createResponseMatchRegExp(request, data) {
    return procedure_1.createResponse(request.id, request.requestType, data);
}
exports.createResponseMatchRegExp = createResponseMatchRegExp;
