// regexp-worker/lib/Procedures/procSleep.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createResponseSleep = exports.createRequestSleep = exports.procSleep = exports.isSleepResponse = exports.isSleepRequest = exports.typeSleep = void 0;
const procedure_1 = require("./procedure");
exports.typeSleep = 'Sleep';
exports.isSleepRequest = procedure_1.genIsRequest(exports.typeSleep);
exports.isSleepResponse = procedure_1.genIsResponse(exports.typeSleep);
function procSleep(r) {
    if (!exports.isSleepRequest(r))
        return undefined;
    if (!isValid(r)) {
        return Promise.resolve(procedure_1.createErrorResponse(r, 'Empty Sleep Duration'));
    }
    return new Promise(resolve => {
        const response = createResponseSleep(r, r.data);
        setTimeout(() => resolve(response), r.data.durationMs);
    });
}
exports.procSleep = procSleep;
function createRequestSleep(data) {
    data = typeof data === 'number' ? { durationMs: data } : data;
    return procedure_1.createRequest(exports.typeSleep, data);
}
exports.createRequestSleep = createRequestSleep;
function createResponseSleep(request, data) {
    return procedure_1.createResponse(request.id, request.requestType, data);
}
exports.createResponseSleep = createResponseSleep;
function isValid(request) {
    return !!(request.data && typeof request.data.durationMs === 'number');
}
