// regexp-worker/lib/Procedures/procEcho.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createResponseEcho = exports.createRequestEcho = exports.procEcho = exports.isEchoResponse = exports.isEchoRequest = exports.typeEcho = void 0;
const procedure_1 = require("./procedure");
exports.typeEcho = 'Echo';
exports.isEchoRequest = procedure_1.genIsRequest(exports.typeEcho);
exports.isEchoResponse = procedure_1.genIsResponse(exports.typeEcho);
function procEcho(r) {
    if (!exports.isEchoRequest(r))
        return undefined;
    if (typeof r.data !== 'string') {
        return procedure_1.createErrorResponse(r, 'Empty Echo');
    }
    return createResponseEcho(r, r.data);
}
exports.procEcho = procEcho;
function createRequestEcho(data) {
    return procedure_1.createRequest(exports.typeEcho, data);
}
exports.createRequestEcho = createRequestEcho;
function createResponseEcho(request, data) {
    return procedure_1.createResponse(request.id, request.requestType, data);
}
exports.createResponseEcho = createResponseEcho;
