// regexp-worker/lib/Procedures/procExecRegExpMatrix.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createResponseExecRegExpMatrix = exports.createRequestExecRegExpMatrix = exports.procExecRegExpMatrix = exports.isExecRegExpMatrixResponse = exports.isExecRegExpMatrixRequest = exports.requestTypeExecRegExpMatrix = void 0;
const procedure_1 = require("./procedure");
const evaluateRegExp_1 = require("../helpers/evaluateRegExp");
exports.requestTypeExecRegExpMatrix = 'ExecRegExpMatrix';
exports.isExecRegExpMatrixRequest = procedure_1.genIsRequest(exports.requestTypeExecRegExpMatrix);
exports.isExecRegExpMatrixResponse = procedure_1.genIsResponse(exports.requestTypeExecRegExpMatrix);
function procExecRegExpMatrix(r) {
    if (!exports.isExecRegExpMatrixRequest(r))
        return undefined;
    const { regExpArray, textArray } = r.data;
    return createResponseExecRegExpMatrix(r, evaluateRegExp_1.execRegExpMatrix(regExpArray, textArray));
}
exports.procExecRegExpMatrix = procExecRegExpMatrix;
function createRequestExecRegExpMatrix(data) {
    return procedure_1.createRequest(exports.requestTypeExecRegExpMatrix, data);
}
exports.createRequestExecRegExpMatrix = createRequestExecRegExpMatrix;
function createResponseExecRegExpMatrix(request, data) {
    return procedure_1.createResponse(request.id, request.requestType, data);
}
exports.createResponseExecRegExpMatrix = createResponseExecRegExpMatrix;
