// regexp-worker/lib/Procedures/procMatchRegExpArray.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createResponseMatchRegExpArray = exports.createRequestMatchRegExpArray = exports.procMatchRegExpArray = exports.isMatchRegExpArrayResponse = exports.isMatchRegExpArrayRequest = exports.requestTypeMatchRegExpArray = void 0;
const procedure_1 = require("./procedure");
const evaluateRegExp_1 = require("../helpers/evaluateRegExp");
exports.requestTypeMatchRegExpArray = 'MatchRegExpArray';
exports.isMatchRegExpArrayRequest = procedure_1.genIsRequest(exports.requestTypeMatchRegExpArray);
exports.isMatchRegExpArrayResponse = procedure_1.genIsResponse(exports.requestTypeMatchRegExpArray);
function procMatchRegExpArray(r) {
    if (!exports.isMatchRegExpArrayRequest(r))
        return undefined;
    try {
        const regex = r.data.regexps.map(r => evaluateRegExp_1.toRegExp(r));
        const regexResult = evaluateRegExp_1.matchRegExpArray(r.data.text, regex);
        return createResponseMatchRegExpArray(r, regexResult);
    }
    catch (e) {
        return procedure_1.createErrorResponse(r, e.toString());
    }
}
exports.procMatchRegExpArray = procMatchRegExpArray;
function createRequestMatchRegExpArray(data) {
    return procedure_1.createRequest(exports.requestTypeMatchRegExpArray, data);
}
exports.createRequestMatchRegExpArray = createRequestMatchRegExpArray;
function createResponseMatchRegExpArray(request, data) {
    return procedure_1.createResponse(request.id, request.requestType, data);
}
exports.createResponseMatchRegExpArray = createResponseMatchRegExpArray;
